lib/src/pubspec.dart

```diff
@@ -635,3 +690,63 @@ class Pubspec extends PubspecBase {
 /// This is slightly more complicated than just checking if the scheme is
 /// 'file', since relative URIs also refer to the filesystem on the VM.
 bool _isFileUri(Uri uri) => uri.scheme == 'file' || uri.scheme == '';
+
+/// Merges [overrides] from an overrides file into a [pubspec] file.
+///
+/// The merging is performed at the stage where the pubspec and overrides files
+/// have both been parsed into [YamlMap]s. Source references of [YamlNode]s are
+/// preserved, so that error messages will point to the correct source location.
+///
+/// Only the the following top-level fields of a pubspec are merged:
+///
+/// - `dependencies`
+/// - `dev_dependencies`
+/// - `dependency_overrides`
+///
+/// All other fields of an overrides file are ignored.
+YamlMap _mergePubspecOverrides(YamlMap pubspec, YamlMap overrides) =>
+    OverrideYamlMap(pubspec, overrides, (key, pubspecField, overridesField) {
+      const dependencyKeys = [
+        'dependencies',
+        'dev_dependencies',
+        'dependency_overrides'
+      ];
+
+      if (!dependencyKeys.contains(key)) {
+        // Only dependencies can be overridden, currently.
+        return pubspecField;
+      }
+
+      return _mergeDependencyOverrides(pubspecField, overridesField);
+    });
+
+/// Merges a dependency section form an overrides file into an existing section
+/// of a pubspec file.
+YamlNode _mergeDependencyOverrides(
```

Comment:
I think we should not merge overrides.
If there is a `pubspec_overrides.yaml` we should only use those overrides, ignoring the ones from pubspec.yaml.

I think that leads to a much simpler mental model

Reply:
Also if we merge them we would need for a way to remove overrides

Reply:
You're right, we should only support `dependency_overrides` in `pubspec_overrides.yaml`.

Reply:
Misunderstood that comment first, but get it now. `dependency_overrides` from `pubspec_overrides.yaml` now fully replace existing `dependency_overrides` in `pubspec.yaml`.